addons/account_check_printing/tests/test_print_check.py

```diff
@@ -347,3 +349,23 @@ def test_multiple_payments_check_number_uniqueness(self):
         move_names = payments.move_id.line_ids.mapped('name')
         self.assertIn(f"Checks - 10001: {payments[0].memo}", move_names)
         self.assertIn(f"Checks - 10002: {payments[1].memo}", move_names)
+
+    def test_number_exceeds_int32_limit(self):
+        """Numbers greater than 2,147,483,647 should raise a ValidationError."""
+        self.journal = self.env['account.journal'].create({
+            'name': 'Test Bank Journal',
+            'type': 'bank',
+            'code': 'TBJ',
+            'bank_statements_source': 'manual',
+            'check_manual_sequencing': True,
+        })
+
+        check_number_too_big = str(2_147_483_648)
+        check_number_normal = str(2_147_483_647)
+        with self.assertRaisesRegex(ValidationError, re.escape(
+                "The check number you entered (2147483648) exceeds the maximum allowed value of 2147483647. Please enter a smaller number."
+            )
+        ):
```

Comment:
The expected regex doesn't have to be the full sentence, just enough words to identify which ValidationError is it from the few key words of its message.
```suggestion
        with self.assertRaisesRegex(ValidationError, "The check number you entered .* exceeds the maximum allowed value"):